### PR TITLE
feat(likes):like/unlike an accommodation facility

### PR DESCRIPTION
**Pull request by alaintresor**
#### What does this PR do?
like or unlike an accommodation facility

#### Description of Task to be completed?

- like an accommodation
- unlike an accommodation
- get all an accommodation likes

#### How should this be manually tested?
After running the app with` npm run start:dev` and  use this endpoint `/api/v1/accommodation/like/[ACCOMMODATION ID]` on a PUT request to Like or Unlike accommodation or GET `/api/v1/accommodation/like/[ACCOMMODATION ID] ` to view all trip requests
OR
after starting a server, you can open any web browser and put this URL `localhost:[YOUR PORT]/api/v1/docs` to test it with swagger
#### What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/story/show/182201728

### CODE REVIEW

**Review by uwambajeddy — DISMISSED** [review]
LGTM

File: src/controllers/accommodation.controller.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
+import db from '../database/models/index.js';
+const {accommodations, UserAccommodation} = db;
+
+
+export const updateLike = async (req, res) => {
+	const userId=req.user.dataValues.id
+	const accommodationId=req.params.id
+	console.log(userId);
```
Comment (Patricknshimiyimana) [inline src/controllers/accommodation.controller.js]: You may wanna remove this console.log

File: src/controllers/accommodation.controller.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+       res.status(400).json({message:"Accommodation not found"})
+      }
+
+      const like = await UserAccommodation.findOne(
+        {where:{accommodationId,
+        userId}}
+      );
+
+		if (like) {
+			console.log(like.dataValues.like);
```
Comment (Patricknshimiyimana) [inline src/controllers/accommodation.controller.js]: And this one too

**Review by Patricknshimiyimana — CHANGES_REQUESTED** [review]

**Review by mwisemarierose — DISMISSED** [review]

**Review by Patricknshimiyimana — DISMISSED** [review]

File: src/controllers/accommodation.controller.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
+import db from '../database/models/index.js';
```
Comment (joshwambere) [inline src/controllers/accommodation.controller.js]: can you follow the naming the team is using (camel case)

File: src/controllers/accommodation.controller.js [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+      if (!accommodation) {
+       res.status(400).json({message:"Accommodation not found"})
+      }
+
+      const like = await UserAccommodation.findOne(
+        {where:{accommodationId,
+        userId}}
+      );
+
+		if (like) {
```
Comment (joshwambere) [inline src/controllers/accommodation.controller.js]: linting needed here

File: src/controllers/accommodation.controller.js [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+				});
+				res.status(200).json({ message: 'Like removed' });
+			}else
+			{
+				await UserAccommodation.update({
+					like: true 
+				},{
+					where:{accommodationId,userId}
+				});
+				res.status(200).json({ message: 'Like added' });
```
Comment (joshwambere) [inline src/controllers/accommodation.controller.js]: may return `accommodation liked`

**Review by joshwambere — CHANGES_REQUESTED** [review]

**Review by uwambajeddy — DISMISSED** [review]

**Review by Placide123 — CHANGES_REQUESTED** [review]
Hi, I did the test but you have to put the limitation not all people can do like/unlike for example like and unlike to a manager or travel admin role is not needed, what I think only client who requested the accommodation can like/unlike. Thanks.

**Review by Placide123 — DISMISSED** [review]

**Review by uwambajeddy — DISMISSED** [review]

File: src/database/models/useraccommodation.js [hunk 6]
```diff
@@ -0,0 +1,34 @@
+import { Model } from 'sequelize';
```
Comment (joshwambere) [inline src/database/models/useraccommodation.js]: can you name this model to make more sense may be likes or something

**Review by joshwambere — DISMISSED** [review]

**Review by Patricknshimiyimana — APPROVED** [review]

**Review by Patricknshimiyimana — APPROVED** [review]

**Review by Placide123 — APPROVED** [review]

**Pull request merged**